src/Stratis.Bitcoin.Features.Api/ApiSettings.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using Stratis.Bitcoin.Configuration;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Api
+{
+    /// <summary>
+    /// Configuration related to the API interface.
+    /// </summary>
+    public class ApiSettings
+    {
+        /// <summary>URI to node's API interface.</summary>
+        public Uri ApiUri { get; set; }
+
+        /// <summary>The callback used to override/constrain/extend the settings provided by the Load method.</summary>
+        private Action<ApiSettings> callback = null;
+
+        /// <summary>
+        /// Constructs this object.
+        /// </summary>
+        public ApiSettings()
+        {
+        }
+
+        /// <summary>
+        /// Constructs this object whilst providing a callback to override/constrain/extend 
+        /// the settings provided by the Load method.
+        /// </summary>
+        /// <param name="callback">The callback used to override/constrain/extend the settings provided by the Load method.</param>
+        public ApiSettings(Action<ApiSettings> callback)
+            : this()
+        {
+            this.callback = callback;
+        }
+
+        /// <summary>
+        /// Constructs this object from the NodeSettings and the provided callback.
+        /// </summary>
+        /// <param name="nodeSettings">The NodeSettings object.</param>
+        /// <param name="callback">The callback used to override the node settings.</param>
+        public ApiSettings(NodeSettings nodeSettings, Action<ApiSettings> callback = null)
+            : this(callback)
+        {
+            this.Load(nodeSettings);
+        }
+
+        /// <summary>
+        /// Loads the API related settings from the application configuration.
+        /// </summary>
+        /// <param name="nodeSettings">Application configuration.</param>
+        public void Load(NodeSettings nodeSettings)
+        {
+            TextFileConfiguration config = nodeSettings.ConfigReader;
+
+            this.ApiUri = config.GetOrDefault("apiuri", new Uri($"http://localhost:{ (nodeSettings.Network.IsBitcoin() ? 37220 : 37221) }"));
```

Comment:
personal opinion - if any more complex constructs are used in interpolated strings, the whole purpose of the interpolated string is thrown out of the window. if you agree, you might want to introduce a variable `port` here and then write

`$"http://localhost:{port}"`

which is very readable (and i think that was the goal with interpolated strings)

alternatively, you can do very well without port variable with old string format:
`string.Format("http://localhost:{0}", nodeSettings.Network.IsBitcoin() ? 37220 : 37221)`

which is imho also more readable than the interpolated string you have there now

but i would prefer the variable with interpolation over string format

Reply:
Changed to use `$"http://localhost:{port}"` to improve readability.